packages/x-components/src/x-modules/url/store/__tests__/actions.spec.ts

```diff
@@ -1,137 +1,26 @@
+import { createUrlStore } from './utils';
 
 describe('testing Url module actions', () => {
   const store = createUrlStore();
-  // eslint-disable-next-line @typescript-eslint/unbound-method
-  const historyReplaceFn = window.history.replaceState;
-
-  beforeEach(() => {
-    window.history.replaceState = historyReplaceFn;
-    window.history.replaceState({}, document.title, window.location.hostname);
-  });
-
-  describe('updateUrl', () => {
-    it('should add the correct params to the url', async () => {
-      resetUrlStateWith(store, {
-        config: {
-          urlParamNames: {
-            query: 'q',
-            relatedTag: 'tag'
-          }
-        },
-        params: {
-          query: 'sudadera',
-          filters: [],
-          relatedTag: ['con capucha', 'disney'],
-          page: 1
-        },
-        extraParams: { warehouse: '0123999' }
-      });
-
-      await store.dispatch('updateUrl');
-
-      expect(window.location.search).toEqual(
-        '?q=sudadera&tag=con+capucha&tag=disney&warehouse=0123999'
-      );
-    });
-
-    // eslint-disable-next-line max-len
-    it('should remove all the parameters from the url that are empty or not valid in the state', async () => {
-      resetUrlStateWith(store, {
-        params: {
-          page: 2,
-          query: 'doramion'
-        }
-      });
-
-      await store.dispatch('updateUrl');
-
-      resetUrlStateWith(store, { params: { page: 1, query: '' } });
-
-      await store.dispatch('updateUrl');
-
-      expect(window.location.search).toEqual('');
-    });
-
-    it('should push the state when the parameter is pushable', async () => {
-      window.history.pushState = jest.fn();
-
-      resetUrlStateWith(store, {
-        config: {
-          urlParamNames: {
-            query: 'q',
-            relatedTag: 'tag'
-          }
-        },
-        params: {
-          query: 'sudadera',
-          filters: [],
-          relatedTag: [],
-          page: 1,
-          scroll: 2
-        }
-      });
-
-      await store.dispatch('updateUrl');
-
-      // eslint-disable-next-line @typescript-eslint/unbound-method
-      expect(window.history.pushState).toHaveBeenCalled();
-    });
-
-    it('should replace the state when the parameter is not pushable', async () => {
-      window.history.replaceState = jest.fn();
-
-      resetUrlStateWith(store, {
-        config: {
-          urlParamNames: {
-            query: 'q',
-            relatedTag: 'tag'
-          }
-        },
-        params: {
-          query: 'sudadera',
-          filters: [],
-          relatedTag: [],
-          page: 1,
-          scroll: 0
-        }
-      });
-
-      await store.dispatch('updateUrl');
-
-      // eslint-disable-next-line @typescript-eslint/unbound-method
-      expect(window.history.replaceState).toHaveBeenCalled();
-    });
-  });
 
   describe('updateStoreFromUrl', () => {
     it('should update the state with the correct url parameters', async () => {
```

Comment:
Why did you remove 3 tests (push, replace an invalid params in the url)? I'd rollback and leave them.

Reply:
The remove emtpy values is responsibility of the getter, I added a test for it there